### ISSUE TITLE
Add api versioning in url and docs like V1 etc

### ISSUE THREAD

**Issue by websoftwares** [LateRegistration/lr-restify#40]

**Comment by rodislav** [comment]
Hey @bafxyz and @websoftwares 

I think the most practical and easy way to do it to add version in the url
but the question remains - how to handle it in cod ? :))) 
we could have:
1. versioned controllers, eg. bus.controller.v1.js // I do not really like it
1. universal controllers, but versioned functions, eg. function get_v1() // mmm... so so
1. universal controllers, universal functions, first param is the version, eg. function get(version, ...) // this looks like a nice way to handle it

what do you think ?

**Comment by websoftwares** [comment]
@bafxyz and @rodislav i think we should extract the version number from url and have the routing decide to what file/controller it should go. so separated files per action is what i like  most following  the [ADR](https://github.com/pmjones/adr) so we can have folder with V1/controller and V2/controller we ned to find a way to extend/inherent  from an abstract controller for the shared common logic.

**Comment by rodislav** [comment]
you mean something like this ? https://bitbucket.org/rodislav/storemanager/src/1fe901aba721abb9efece1593f5e44f62003ea1f/src/main/java/com/storemanager/features/?at=master

there I have all files separated per features (some kind of actions), 
the entry points are not considered part of a feature, like controllers, and only they are versioned: https://bitbucket.org/rodislav/storemanager/src/1fe901aba721abb9efece1593f5e44f62003ea1f/src/main/java/com/storemanager/main/http/?at=master

so basically I version only the api, not the business logic

**Comment by websoftwares** [comment]
Something like your comment but, also like this one :-)

https://github.com/websoftwares/example-webapp/tree/master/application

**Comment by rodislav** [comment]
[this thing:](https://github.com/pmjones/adr#adr-revision) is totally the same as [this one](http://dotnetslackers.com/images/articleimages/build-a-front-end-of-an-ecommerce-system-using-aspnet-mvc-framework--part-1-2.gif) - is still layered, so I do not see any difference, extra folders makes no sence (with 1-2 files per folder, most of times 1)

is easier to have all files for a group of business cases in one folder, and if the files grow in number, they should be grouped in folders --> but this could mean, that business features is too big so it should be split in smaller features.

I'll create a branch with example, so we can see there how to proceed

**Comment by rodislav** [comment]
check this branch https://github.com/LateRegistration/lr-restify/tree/project-structure-demo/src

**Comment by rodislav** [comment]
finally, the template for organising repo is this branch: https://github.com/LateRegistration/lr-restify/tree/project-structure-demo/src

**Comment by websoftwares** [comment]
Please remove the assets these should be embedded in the mongo image and only process maybe we even should move the batch part (parsers) to mongo machine since  they are only used on that image.

**Comment by rodislav** [comment]
yeah, good idea, I'll move them
